Tests/MessagingInApp/Views/InAppMessageViewTest.swift

```diff
@@ -664,212 +668,108 @@ class InAppMessageViewTest: IntegrationTest {
+        // ignore the elementid and always return the same message.
+        // So, this test wants each inline View to have a separate message but instead each view will get the same message.
+        // PR that would fix this test: https://github.com/customerio/customerio-ios/pull/776
 
-        // Tap the button again and verify that
-        // the call to onActionClick updates
-        onCustomActionButtonPressed(onInlineView: inlineView)
+        let givenInlineMessage1 = Message.randomInline
+        let givenInlineMessage2 = Message.randomInline
+        queueMock.getInlineMessagesReturnValue = [givenInlineMessage1, givenInlineMessage2]
 
-        XCTAssertTrue(inlineMessageDelegateMock.onActionClickCalled)
-        XCTAssertEqual(inlineMessageDelegateMock.onActionClickCallsCount, 2)
-        XCTAssertFalse(eventListenerMock.messageActionTakenCalled)
-    }
+        let inlineView1 = InAppMessageView(elementId: givenInlineMessage1.elementId!)
+        let inlineView2 = InAppMessageView(elementId: givenInlineMessage2.elementId!)
 
-    @MainActor
-    func showInlineMessageForMetrics(setDelegate: Bool = false) async -> InAppMessageView {
-        messagingInAppImplementation.setEventListener(eventListenerMock)
-        messagingInAppImplementation.setEventBusHandler(eventBusHandlerMock)
+        await onDoneRenderingInAppMessage(givenInlineMessage1, insideOfInlineView: inlineView1)
 
-        let givenInlineMessage = Message.randomInline
-        queueMock.getInlineMessagesReturnValue = [givenInlineMessage]
+        assert(message: givenInlineMessage1, didCallMessageShownEventListener: true)
+        assert(message: givenInlineMessage2, didCallMessageShownEventListener: false)
 
-        let inlineView = InAppMessageView(elementId: givenInlineMessage.elementId!)
-        if setDelegate {
-            inlineView.onActionDelegate = inlineMessageDelegateMock
-        }
-        await onDoneRenderingInAppMessage(givenInlineMessage, insideOfInlineView: inlineView)
+        await onDoneRenderingInAppMessage(givenInlineMessage2, insideOfInlineView: inlineView2)
 
-        return inlineView
+        assert(message: givenInlineMessage1, didCallMessageShownEventListener: true)
+        assert(message: givenInlineMessage2, didCallMessageShownEventListener: true)
     }
 }
 
 @MainActor
 extension InAppMessageViewTest {
-    func onCustomActionButtonPressed(onInlineView inlineView: InAppMessageView) {
+    func onCustomActionButtonPressed(onInlineView inlineView: InAppMessageView) async {
         // Triggering the custom action button on inline message from the web engine
         // mocks the user tap on custom action button
         getWebEngineForInlineView(inlineView)?.delegate?.tap(name: "", action: "Test", system: false)
+
+        // when a button is pressed, an eventbus event is posted for metrics tracking. Return function after the tracked event is posted.
+        await waitForEventBusEventsToPost()
```

Comment:
With not mocking the eventbus, we need to add this wait so check if push metrics were reported after an event was posted to the eventbus.